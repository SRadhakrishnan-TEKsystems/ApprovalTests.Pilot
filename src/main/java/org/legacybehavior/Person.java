package org.legacybehavior;

import java.time.Instant;
import java.util.UUID;

public class Person {
    private final String name;
    private final String superHeroName;
    private final String universe;
    private final UUID id;
    private final Instant creationInstance;

    private Person(String name, String superHeroName, String universe, UUID id, Instant creationInstance) {
        this.name = name;
        this.superHeroName = superHeroName;
        this.universe = universe;
        this.id =id;
        this.creationInstance = creationInstance;
    }

    public static Person getInstance(String name, String superHeroName, String universe, UUID id, Instant creationInstance) {
        return new Person(name,superHeroName,universe,id,creationInstance);
    }
}
