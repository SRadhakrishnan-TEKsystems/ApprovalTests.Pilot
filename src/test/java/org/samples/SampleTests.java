package org.samples;


import com.google.gson.GsonBuilder;
import org.approvaltests.Approvals;
import org.approvaltests.JsonApprovals;
import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.namer.NamedEnvironment;
import org.approvaltests.namer.NamerFactory;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;
import org.legacybehavior.Person;

import java.awt.*;
import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SampleTests
{
  @Test
  public void testNormalJunit()
  {
    assertEquals(5, 5);
  }
  @Test
  public void testStrings()
  {
    Approvals.verify("Hello World!");
  }
  @Test
  public void testArrays()
  {
    String[] names = {"Llewellyn", "James", "Dan", "Jason", "Katrina"};
    Arrays.sort(names);
    Approvals.verifyAll("", names);
  }

  @Test
  public void testObject() {
    Rectangle rectangle = new Rectangle(5, 5, 5, 7);
    Approvals.verify(rectangle.toString());
  }

  @Test
  public void testJsonApprovals() {
    Person hero = Person.getInstance("Tony Stark", "Iron Man", null,null, null);
    JsonApprovals.verifyAsJson(hero, GsonBuilder::serializeNulls);
  }

  @Test
  public void testCombinationApprovals() {
    Integer[] lengths = new Integer[]{4, 5, 6};
    String[] words = new String[]{"Bookkeeper", "applesauce"};
    CombinationApprovals.verifyAllCombinations((i, s)->s.substring(0,i),lengths,words);
  }

  @ParameterizedTest
  @ValueSource(strings = {"parameter1", "parameter2"})
  void sampleParameterizedTest(String parameter)
  {
    try (NamedEnvironment en = NamerFactory.withParameters(parameter))
    {
      // your code goes here
      Object output = parameter;
      Approvals.verify(output);
    }
  }

}
