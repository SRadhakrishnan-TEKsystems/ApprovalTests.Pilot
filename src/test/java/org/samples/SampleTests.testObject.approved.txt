java.awt.Rectangle[x=5,y=5,width=5,height=7]